wsd/debug-main.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+#include "config.h"
+#include "StringVector.hpp"
+#include "Util.hpp"
+#include "Log.hpp"
+#include "Common.hpp"
+#include "future"
+
+extern "C" int forkit_main(int argc, char **argv);
+
+char** convertStringVectorToCharArray(StringVector stringVector)
+{
+    char** charArray = new char*[stringVector.size()];
+
+    for (size_t i = 0; i < stringVector.size(); ++i) {
+        // Allocate memory for each C-style string and copy the content
+        charArray[i] = new char[stringVector[i].size() + 1]; // +1 for null-terminator
+        std::strcpy(charArray[i], stringVector[i].c_str());
+    }
+
+    return charArray;
+}
+
+extern "C" int createForkit(const std::string forKitPath, const StringVector args)
+{
+    // Create a promise to capture the result from the thread.
+    std::promise<int> resultPromise;
+
+    LOG_INF(forKitPath);
+    char** argv = convertStringVectorToCharArray(args);
+
+    std::thread ([&] {
+        Util::setThreadName("debug main kit_in_process");
+        int result = forkit_main(args.size(), argv);
+
+        // Fulfill the promise with the result.
+        resultPromise.set_value(result);
+    }).detach();
+
+    // Wait for the thread to finish and get the result using a future.
+    std::future<int> resultFuture = resultPromise.get_future();
+    int result = resultFuture.get();
```

Comment:
The proper way of waiting for a thread to finish is to join the thread.

There is no good reason to detach the thread and use a promise/future.
```
int result = 0;
std::thread worker = std::thread ([&] {
        Util::setThreadName("debug main kit_in_process");
        result = forkit_main(args.size(), argv);
    });

worker.join();
return result;
```